graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/builtins/GlobalBuiltins.java

```diff
@@ -1284,6 +1300,50 @@ public boolean isCallerSensitive() {
         }
     }
 
+    /**
+     * Implementation of Module Block serialize-method
+     */
+    public abstract static class JSGlobalSerializeNode extends JSBuiltinNode {
+
+        protected JSGlobalSerializeNode(JSContext context, JSBuiltin builtin) {
+            super(context, builtin);
+        }
+
+        protected byte[] serialize(Object value) {
```

Comment:
That should be a `@Specialization` method, right? Is `byte[]` the right return type, is that something we can process in the engine? Where could that value flow to?

Reply:
The value needs to be deserialized by the according method which takes a byte-array to produce a module block.